openssl/src/pkey_ctx.rs

```diff
@@ -105,6 +109,21 @@ impl HkdfMode {
     pub const EXPAND_ONLY: Self = HkdfMode(ffi::EVP_PKEY_HKDEF_MODE_EXPAND_ONLY);
 }
 
+/// Nonce type for ECDSA and DSA.
+#[cfg(ossl320)]
+#[derive(Debug, PartialEq)]
+pub struct NonceType(c_uint);
+
+#[cfg(ossl320)]
+impl NonceType {
+    /// This is the default mode. It uses a random value for the nonce k as defined in FIPS 186-4 Section 6.3
+    /// “Secret Number Generation”.
+    pub const RANDOM_K: Self = NonceType(0);
```

Comment:
Does OpenSSL not have a constant for these values?

Reply:
Not that I could find. The [source code](https://github.com/openssl/openssl/blob/da1c088f599af3755aaeed1c447a39621ef12e1f/providers/implementations/signature/ecdsa_sig.c#L198) treats it as a boolean:
```c
    if (ctx->nonce_type != 0) {
        ret = ossl_ecdsa_deterministic_sign(tbs, tbslen, sig, &sltmp,
                                            ctx->ec, ctx->nonce_type,
                                            ctx->mdname,
                                            ctx->libctx, ctx->propq);
    } else {
        ret = ECDSA_sign_ex(0, tbs, tbslen, sig, &sltmp, ctx->kinv, ctx->r,
                            ctx->ec);
```